cpp/include/cuml/fil/fil.h

```diff
@@ -224,6 +224,9 @@ struct forest_params_t {
   // labels in leaves instead of the whole vector, this keeps track
   // of the number of classes
   int num_classes;
+  // blocks_per_sm, if set, works as a limit to improve cache hit rate for larger forests
```

Comment:
It would be better to explain what special values are there for `blocks_per_sm`. E.g. "if 0, launch ceildiv(num_rows, NITEMS) blocks".